text/0054-rth-payment-id-memos.md

```diff
@@ -0,0 +1,127 @@
+
+This proposed Destination With Payment Request Id Memo fixes this by holding an 8-byte payment request ID number. If Alice wants to fulfill a payment request from Bob, Alice can attach a sender with payment request ID memo to Alice's main TxOut being sent to Bob. Alice can then attach a destination with payment request ID memo (containing the same request ID) to the change TxOut. This allows Alice to have her own record of fulfilling the payment request.
+
+## Authenticated Sender and Destination With Payment Id Memos
+
+For clients wishing to implement the opposite flow, the sender and destination with payment ID memos are proposed (type bytes `0x0102` and `0x0204`, respectively). Suppose Alice wishes to send a payment to Bob, but Bob either does not have a public address or has not shared it with her. Alice can, using some shared service or communication channel, notify Bob that a payment with a specific ID is waiting for him. Bob may then wish to create and/or share his public address with Alice to receive the payment.
+
+With all of this established, Alice is able to then create and send the payment to Bob. Alice should build a sender with payment ID memo and attach it to the TxOut for Bob. Paired with that, for her own record, she should attach a destination with payment ID memo to her change TxOut so she also has record of this payment intent being fulfilled. Once Bob receives the payment, he can see that it contains a sender memo with the same payment ID as the previously shared payment intent. This informs Bob that the pending payment to him is complete. Had Bob received a different payment (from Alice or another party) in between the time the intent was shared and completed, he would know that this is not related to the same intent.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## 0x0203 Destination With Payment Request Id Memo
+
+The 64-byte memo data is laid out as follows:
+
+| Byte range | Item |
+| ---------- | ---- |
+| 0 - 16     | Recipient's Address Hash |
+| 16 - 17    | The number of recipients, as an unsigned 8-bit number |
+| 17 - 24    | Big-endian bytes of fee amount, as an unsigned 56-bit number |
+| 24 - 32    | Big-endian bytes of the total outlay amount, as an unsigned 64-bit number |
+| 32 - 40    | Big-endian bytes of 8-byte payment request id number |
+| 40 - 64    | Unused bytes                  |
+
+This is exactly identical to RTH's destination memo except it contains an 8-byte payment request ID number. A user fulfilling a payment request will use this memo and populate that field with the same ID they put into an authenticated sender with payment request ID memo. This will allow the fulfilling user to keep a record that the payment request was fulfilled.
```

Comment:
```suggestion
This is RTH's 0x0200 Destination Memo plus an 8-byte payment request ID. A user fulfilling a payment request will use this memo and populate that field with the same ID they put into an Authenticated Sender With Payment Request ID Memo. This will allow the fulfilling user to keep a record that the payment request was fulfilled.
```